components/fonts/shaper.rs

```diff
@@ -107,16 +108,17 @@ impl ShapedGlyphData {
             let x_advance = Au::from_f64_px(x_advance);
             let y_advance = Au::from_f64_px(y_advance);
 
-            let offset = if x_offset == Au(0) && y_offset == Au(0) && y_advance == Au(0) {
-                None
-            } else {
-                // adjust the pen..
-                if y_advance > Au(0) {
-                    *y_pos -= y_advance;
-                }
+            let offset =
+                if x_offset == Au::zero() && y_offset == Au::zero() && y_advance == Au::zero() {
```

Comment:
```suggestion
                if x_offset.is_zero() && y_offset.is_zero() && y_advance.is_zero() {
```

Reply:
This function is present in `num_traits`'s `Zero` trait, but I'm using `euclid`'s `Zero` trait which doesn't export `.is_zero()`. Should I add `num_traits`/the feature flag in `app_units` instead by modifying `Cargo.toml` or leave it as is?

Reply:
Perhaps I should use `style::Zero` instead?

Reply:
Ah, yeah, that's annoying, you can leave it as is

Reply:
Hrm. You should import `num_traits::Zero` here instead of the `euclid` version for this to work properly. `Au` does implement that and it includes `is_zero()`.

Reply:
> Hrm. You should import `num_traits::Zero` here instead of the `euclid` version for this to work properly. `Au` does implement that and it includes `is_zero()`.

It does, but `components/fonts` doesn't have the `num-traits` dependency inside its `Cargo.toml`, so I'd need to add it before using that trait IIRC. If that's ok to add a dependency here though I can do that

Reply:
It's okay to do that. We already depend on `num-traits` so this isn't adding else to the build chain other than perhaps modifying the crate build order (though unlikely).

Reply:
Ok, it's done now